src/core/line/line.ts

```diff
@@ -0,0 +1,44 @@
+import {
+    Directive,
+    Renderer,
+    ElementRef,
+    QueryList
+} from '@angular/core';
+
+/* Shared directive to count lines inside text */
+@Directive({ selector: '[md-line]' })
+export class MdLine {}
+
+/* Service that takes a query list of lines and sets the correct class on the host */
+export class MdLineSetter {
+  constructor(private _lines: QueryList<MdLine>, private _renderer: Renderer,
+              private _element: ElementRef) {
+    this._init();
```

Comment:
Just hate having mess in the constructor, but given that it would need a method description... removing :D